api/apps/v1alpha1/nemo_common_types.go

```diff
@@ -76,3 +76,68 @@ type DatabaseCredentials struct {
 	// +kubebuilder:default:="password"
 	PasswordKey string `json:"passwordKey,omitempty"`
 }
+
+// WandBSecret represents the secret and key details for the Weights and Biases service.
+type WandBSecret struct {
+	// Name is the name of the Kubernetes Secret containing the WandB API key.
+	// Required, must not be empty.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name,omitempty"`
```

Comment:
since we have the `+kubebuilder:validation:Required`, we should probably not have `omitempty` in the json tag

Reply:
yes, makes sense, this is fine if we were assigning a default value like others, but otherwise omitempty needs to be dropped. will update.